### ISSUE TITLE
Project cannot be built

### ISSUE THREAD

**Issue by Artur-** [alloc/vite-plugin-compress#11]
Running `yarn build` results in
```
src/plugin.ts:138:14 - error TS2551: Property 'resolveBuiltUrl' does not exist on type 'Plugin'. Did you mean 'resolveFileUrl'?

138         this.resolveBuiltUrl = url => {
                 ~~~~~~~~~~~~~~~

src/plugin.ts:138:32 - error TS7006: Parameter 'url' implicitly has an 'any' type.

138         this.resolveBuiltUrl = url => {
                                   ~~~

src/plugin.ts:168:55 - error TS2571: Object is of type 'unknown'.

168               debug(`Failed to transform "${id}". ` + err.message)
                                                          ~~~

```

The `Object is of type 'unknown'` is related to TypeScript 4.4 and easy to fix but what is `resolveBuiltUrl` supposed to do? It looks like dead code

**Comment by aleclarson** [comment]
See here https://github.com/alloc/vite-plugin-compress/issues/3#issuecomment-927998477

**Comment by Artur-** [comment]
That says it is fixed in the `next` branch but that fails in the same way. The code is the same https://github.com/alloc/vite-plugin-compress/blob/next/src/plugin.ts#L133

**Comment by aleclarson** [comment]
> the fix depends on [#1675](https://github.com/vitejs/vite/pull/1675), which I haven't had time to write tests for.

Unfortunately, the `next` branch depends on an unmerged Vite PR.

**Comment by Artur-** [comment]
Ah, now I see. I guess there is no chance of reverting the change in this plugin until the Vite PR is done so this project could compile?
